mobile/src/stateMachine/views/qol/qolQuestion.tsx

```diff
@@ -0,0 +1,199 @@
+import { ViewState } from '../base';
+import React from 'react';
+import AppViewModel from 'src/viewModels';
+import { observer } from 'mobx-react';
+import { StyleSheet, Text, View, Dimensions, Animated } from 'react-native';
+import { MasloPage, Container, Button } from 'src/components';
+import { ScenarioTriggers } from '../../abstractions';
+import Colors from '../../../constants/colors';
+import TextStyles from '../../../../src/styles/TextStyles';
+import { QolSurveyType } from 'src/constants/QoL';
+
+const minContentHeight = 560;
+
+@observer
+export class QolQuestion extends ViewState {
+
+    labelState = {
+        opacity: new Animated.Value(1),
+    };
+
+    constructor(props) {
+        super(props);
+        this._contentHeight = this.persona.setupContainerHeight(minContentHeight, { rotation: -140 - (this.viewModel.domainNum * 30), transition: { duration: 1 }, scale: 0.8 });
+        this.viewModel.origMags = this.persona.qolMags;
+        this.persona.qolMags = this.viewModel.qolMags;
+    }
+
+    public get viewModel() {
+        return AppViewModel.Instance.QOL;
+    }
+
+    async start() {
+        // Not yet implemented
+    }
+
+    private cancel = () => {
+        this.persona.qolMags = this.viewModel.origMags;
+        this.trigger(ScenarioTriggers.Cancel);
+    }
+
+    // Called when survey is completed
+    private finish = async () => {
+        this.trigger(ScenarioTriggers.Submit);
+        await this.viewModel.sendSurveyResults();
+
+        if (this.viewModel.isUnfinished) {
+            await this.viewModel.saveSurveyProgress(null);
+        }
+        if (this.viewModel.QolSurveyType === QolSurveyType.Full) {
+            this.viewModel.updatePendingFullQol();
+        }
```

Comment:
What do you think about making this abstract so that it's just `updatePendingQol` and it will automatically figure out which type of qol it is?

Reply:
I have done just that in #79